src/lib.rs

```diff
@@ -312,28 +332,22 @@ impl KaboodleInner {
                     // ramp down from a 100% chance if we don't know of any other peers yet to a
                     // minimum of 1% once we have some sufficient number.
                     // See Section 3.2 in the SWIM paper for more thoughts on this logic.
-                    // TODO: temporarily commented out because this logic makes large numbers of
-                    // peers more likely to accidentally end up in several sub-meshes, rather than
-                    // one large mesh. More thought required.
-                    // let num_other_peers = known_peers.len() - 2; // minus ourselves and the newcomer
-                    // let percent_chance_of_sending_peers =
-                    //     std::cmp::max(1, 100 - usize::pow(num_other_peers, 2)) as f32 / 100.0;
-                    // let should_send_peers = self.rng.gen::<f32>() < percent_chance_of_sending_peers;
-                    // if !should_send_peers {
-                    //     log::info!("Not sending known peers to new peer");
-                    //     drop(known_peers);
-                    //     continue;
-                    // }
+                    let num_other_peers = known_peers.len() - 2; // minus ourselves and the newcomer
+                    let percent_chance_of_sending_peers =
+                        std::cmp::max(1, 100 - i64::pow(num_other_peers as i64, 2)) as f64 / 100.0;
+                    let should_send_peers = self.rng.gen_bool(percent_chance_of_sending_peers);
```

Comment:
Calling out that in addition to uncommenting this code, I switched to the `gen_bool` function that I hadn't noticed last time.